internal/transport/grpcframer/framer.go

```diff
@@ -0,0 +1,267 @@
+/*
+ *
+ * Copyright 2024 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+// Package grpcframer defines the interface for implementing an HTTP/2 framer,
+// its required types and an implementation of said framer.
+package grpcframer
+
+import "sync"
+
+// FrameType represents the type of a Frame and its value according to the
+// HTTP/2 spec.
+// See https://httpwg.org/specs/rfc7540.html#FrameTypes.
+type FrameType uint8
+
+const (
+	DataFrameType         FrameType = 0x0
+	HeadersFrameType      FrameType = 0x1
+	PriorityFrameType     FrameType = 0x2
+	RSTStreamFrameType    FrameType = 0x3
+	SettingsFrameType     FrameType = 0x4
+	PushPromiseFrameType  FrameType = 0x5
+	PingFrameType         FrameType = 0x6
+	GoAwayFrameType       FrameType = 0x7
+	WindowUpdateFrameType FrameType = 0x8
+	ContinuationFrameType FrameType = 0x9
```

Comment:
Same here too, it might be better to name these `FrameTypeData` and `FrameTypeHeaders` etc.